packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -301,11 +309,35 @@ class CloudWatchLoggerPlugin extends AWSLoggerPlugin
 
   /// Whether a [logEntry] should be logged by this plugin.
   bool _isLoggable(LogEntry logEntry) {
-    if (!_enabled) {
-      return false;
-    }
+    if (!_enabled) return false;
+
     final loggingConstraint = _getLoggingConstraint();
-    return logEntry.level >= loggingConstraint.defaultLogLevel;
+    final hasUserLogLevel = loggingConstraint.userLogLevel.containsKey(_userId);
+    LogLevel? logLevel;
+
+    if (hasUserLogLevel) {
+      final userLogLevel = loggingConstraint.userLogLevel[_userId]!;
+      logLevel =
+          _getCategoryLogLevel(userLogLevel.categoryLogLevel, logEntry) ??
+              userLogLevel.defaultLogLevel;
+    } else {
+      logLevel =
+          _getCategoryLogLevel(loggingConstraint.categoryLogLevel, logEntry);
+    }
+
+    return logEntry.level >= (logLevel ?? loggingConstraint.defaultLogLevel);
+  }
+
+  LogLevel? _getCategoryLogLevel(
```

Comment:
A bit confused by this method - we seem to have a dictionary with all the string mappings that we can access by key, but we're iterating instead each time because we didn't set the values to lower case?  

Is that necessary?  It's seems a bit strange to use a hashmap in this way.